### DISCUSSION TITLE
Key features that is stoping you moving from other API Client?

### DISCUSSION THREAD

**Discussion by rangav** [thunderclient/thunder-client-support#548] [Q&A]
Hi all,

Let me know what are the key features missing that is stoping you moving from other API Clients?

**Comment by JayRSG** [comment]
Looking forward to a Pre-request Script feature available in POSTMAN. 
I am unable to set csrf cookies for a particular request before sending the actual request that requires an xsrf-token.
I have tried to achieve that using test feature of thunderclient but it did not work. There is nothing mentioned in the documentation. 
The pre-request feature in POSTMAN allows us to send request and fetch information from that request which can be used in the current request.

![image](https://user-images.githubusercontent.com/14149138/171993402-a81f3353-db66-4e93-b2d4-b8746086d362.png)

**Reply by ornsio** [reply]
I use this feature in postman extensively to store a handful of useful test inputs for endpoints where the input is part of the base URL string itself (i.e. not a query parameter). I just have several lines of script code that all set the same variable, use comment toggling to pick the one I want on any given execution, and then reference that variable in the URL string. Having this in script form also allows me to very easily annotate each one with a brief description of what it is (such as a human-readable explanation of what a GUID refers to)

**Comment by rangav** [comment]
@JayRSG you can save response cookie to variable and use it other requests 

see details here
https://github.com/rangav/thunder-client-support#setenv

Let me know if you have any other questions

**Reply by JayRSG** [reply]
@rangav  Unfortunately this is not working. I have set the cookie.xsrf-token from one request and set an environment variable with it but while passing it to other request as header I receive csrf token mismatch. This is not an issue in postman as there is a pre-request script that does fetch the csrf-token and then makes a request.

**Reply by rangav** [reply]
@JayRSG can you create a new issue for this with screenshots and I will look into it.

**Reply by JayRSG** [reply]
Sure I will do it @rangav

**Comment by sticl** [comment]
OAuth 2: Authorization code with PKCE.

It's possible in Postman:
https://vmsdurano.com/requesting-access-token-from-azure-b2c-with-pkce-using-postman/

**Reply by rangav** [reply]
It’s in roadmap, there is already existing request for this.
https://github.com/rangav/thunder-client-support/issues/287

you can add your comments to above link.

**Reply by rangav** [reply]
OAuth with PKCE feature is now implemented and published to marketplace, please update to v1.16.6

See all features and bugs fixed released
https://github.com/rangav/thunder-client-support/releases/tag/v1.16.6
